### ISSUE TITLE
Spec references RFC7231 which is obsoleted

### ISSUE THREAD

**Issue by rivantsov** [graphql/graphql-over-http#201]
Here are couple of things I found that might be an issue: 

Section [Response](https://github.com/graphql/graphql-over-http/blob/main/spec/GraphQLOverHTTP.md#response)

> A server must comply with RFC7231.

First, it would be nice to provide RFC title along with numeric ID, just as a nice gesture to the reader. 
Secondly, this RFC page has a note on top: "Obsoleted by: 9110"; should it reference the newer RFC directly? Or some comment is needed. 

Note: If changing ref to 9110, then some other links to its subsections should be updated. Ex: link to "HTTP 1.1 Accept" section.

**Comment by michaelstaib** [comment]
I agree, we should make it easier for the reader and refer to concrete sections that we refer to if possible. In the concrete case it would be content negotiation.

https://datatracker.ietf.org/doc/html/rfc9110/#section-12.4

**Comment by michaelstaib** [comment]
That the specs get out of date sometimes is not so much a problem, in these case we just need to check the changelog to verify and do PR to update. I went over the changes and there is nothing changed that concerns the accept/content-negotiation semantics.

I like the suggestion adding the RFC title. Could be in braces.

**Comment by rivantsov** [comment]
well, links get obsolete over time - that's understandable. But in this case, it's at time of spec publishing. Better be updated i think

**Comment by michaelstaib** [comment]
Sure, the thing here is that we started work on this SPEC in 2019 so references might get obsoleted by the time we are finished with it. Before we ratify it we must go through and update spec links again. We will put an action item on this for when we do the finishing touches to go over spec references and validate / update them.
